### DISCUSSION TITLE
BASIC support?

### DISCUSSION THREAD

**Discussion by c64cryptoboy** [slajerek/RetroDebugger#57] [Ideas]
Support for analyzing a running Commodore V2 BASIC program may not be a popular feature request, but I'd certainly use it. :D
I took a vacation last week, and my wife hurt her knee, so we stayed a lot in the hotel.  During this time, I whipped up a python prototype that analyzes a memory dump for its BASIC code and BASIC variable memory allocations in order to show all the variables, their values, and which BASIC lines number read and which line numbers modify those variables.
I showed its output on three commercial BASIC games:  https://github.com/c64cryptoboy/CommodoreBasicVarAnalysis

That functionality would be so much more useful if rewritten and integrated directly into your debugger -- that said, that may be a lot of work for a minimum amount of user engagement.  If that's so, one (smaller) change would be useful.  CURLIN (16-bit value at $0039 on a C64) holds the currently-executing BASIC line number, so if I could set a 16-bit value breakpoint (when a certain line number is reached, I could then export snapshots for my python to analyze.  Such a capability would have been really useful when I was analyzing the self-learning AI in the C64 BASIC game Fortress (SSI, 1984).

**Comment by og2t** [comment]
A bit OT but there's a [tool online](https://stigc.dk/c64/basic/) that allows pasting BASIC code. Might be useful.
